analyses/10_mofa/11_easier.R

```diff
@@ -67,49 +67,50 @@ getFeat <- function(dataobj,
+      # remove.genes.ICB_proxies = remove.genes.ICB_proxies
     )
 
     # Quantification of TF activity scores with DoRothEA
     TF <- compute_TF_activity(
-      RNA.tpm = dataobj$tpm[[dataset]],
-      remove.genes.ICB_proxies = remove.genes.ICB_proxies
+      RNA_tpm = dataobj$tpm[[dataset]],
+      # remove.genes.ICB_proxies = remove.genes.ICB_proxies
     )
 
     # Storing of the computed features in the output object
     dataobj$cellfrac[[dataset]] <- cellfrac
     dataobj$pathway[[dataset]] <- pathway$scores
     dataobj$tf[[dataset]] <- TF$scores
 
-    # Computation of immune response scores (proxies)
-    proxies <- compute_gold_standards(
-      RNA.tpm =
-        as.data.frame(dataobj$tpm[[dataset]]),
-      list_gold_standards = selscores,
-      cancertype = cancertype
-    )
-    proxies.mat <- as.data.frame(matrix(unlist(proxies),
-      nrow = length(proxies),
-      byrow = T
-    ))
-    rownames(proxies.mat) <- names(proxies)
-    colnames(proxies.mat) <- colnames(proxies[[1]])
-    proxies.mat <- t(proxies.mat)
-
-    # Rotation of the "Chemokines" score to agree with "CYT" direction
-    chemosign <- sign(cor(proxies.mat[, "CYT"], proxies.mat[, "chemokines"]))
-    if (chemosign < 0) proxies.mat[, "chemokines"] <- -proxies.mat[, "chemokines"]
-
-    # Computation of the median scaled response
-    immscoreZ <- immscoreZ[match(selscores, immscoreZ$score), ]
-    response <- proxies.mat[, match(selscores, colnames(proxies.mat))]
-    response <- ((t(response) - immscoreZ$mean) / immscoreZ$sd)
-    response <- apply(response, 2, median)
-    proxies.mat <- as.data.frame(proxies.mat)
-    proxies.mat$response <- response
-
-    dataobj$immresp[[dataset]] <- proxies.mat
+    # # Computation of immune response scores (proxies)
+    # proxies <- compute_gold_standards(
+    #     RNA_tpm =
+    #         as.data.frame(dataobj$tpm[[dataset]]),
+    #     list_gold_standards = selscores,
+    #     cancertype = cancertype
+    # )
+    # proxies.mat <- as.data.frame(matrix(unlist(proxies),
+    #     nrow = length(proxies),
+    #     byrow = T
+    # ))
+    # rownames(proxies.mat) <- names(proxies)
+    # colnames(proxies.mat) <- colnames(proxies[[1]])
+    # proxies.mat <- t(proxies.mat)
+
+    # # Rotation of the "Chemokines" score to agree with "CYT" direction
+    # chemosign <- sign(cor(proxies.mat[, "CYT"], proxies.mat[, "chemokines"]))
+    # if (chemosign < 0) proxies.mat[, "chemokines"] <- -proxies.mat[, "chemokines"]
+
+    # # Computation of the median scaled response
+    # immscoreZ <- immscoreZ[match(selscores, immscoreZ$score), ]
+    # response <- proxies.mat[, match(selscores, colnames(proxies.mat))]
+    # response <- ((t(response) - immscoreZ$mean) / immscoreZ$sd)
+    # response <- apply(response, 2, median)
+    # proxies.mat <- as.data.frame(proxies.mat)
+    # proxies.mat$response <- response
+
+    # dataobj$immresp[[dataset]] <- proxies.mat
```

Comment:
Hi @olapuentesantana, 

can you please advise what would be the appropriate replacement in the latest easier version? It seems the `compute_gold_standards` function got removed.

Reply:
`compute_scores_immune_response` function replaced `compute_gold_standards`.

Now the immune response scores can be computed like this (no need to specify cancer type):
```
scores_immune_response <- compute_scores_immune_response(
   RNA_tpm = as.data.frame(dataobj$tpm[[dataset]],   
   selected_scores = selscores
)
```
The output is now a numeric matrix with samples in rows and published scores (gold standards) in columns, so I think the script should look like this after computing the scores.

```
proxies.mat <- t(proxies.mat)

# Rotation of the "Chemokines" score to agree with "CYT" direction
chemosign <- sign(cor(proxies.mat[, "CYT"], proxies.mat[, "chemokines"]))
if (chemosign < 0) proxies.mat[, "chemokines"] <- -proxies.mat[, "chemokines"]

# Computation of the median scaled response
immscoreZ <- immscoreZ[match(selscores, immscoreZ$score), ]
response <- proxies.mat[, match(selscores, colnames(proxies.mat))]
response <- ((t(response) - immscoreZ$mean) / immscoreZ$sd)
response <- apply(response, 2, median)
proxies.mat <- as.data.frame(proxies.mat)
proxies.mat$response <- response

dataobj$immresp[[dataset]] <- proxies.mat
```

Reply:
The computation of the features looks good.